### DISCUSSION TITLE
If the database password contains ‘@’, Unable to parse the password properly

### DISCUSSION THREAD

**Discussion by Qiii103** [sqlalchemy/sqlalchemy#8122] [Usage Questions] [answered]
Can you modify the RE statement that parses the database URL？
     Such as：
![image](https://user-images.githubusercontent.com/45709293/173016580-298f2341-8566-4656-a8b5-718270bad5ff.png)

**Comment by CaselIT** [comment, marked as answer]
The current regexp is correct.

This is a duplicate of multiple issues: https://github.com/sqlalchemy/sqlalchemy/issues?q=label%3A%22%40+in+password%22

As documented if the user or password contains special char it needs to be escaped https://docs.sqlalchemy.org/en/14/core/engines.html?highlight=password#database-urls

**Reply by Qiii103** [reply]
Ok, thank you！！
